lib/Sema/ConstraintSystem.cpp

```diff
@@ -32,6 +32,40 @@ unsigned ConstraintSystem::getBestSolutionScore(Constraint *constraint)
     return 0;
 }
 
+void ConstraintSystem::mapOverloadChoices(Solution *solution)
+{
+    for (auto &pair : solution->overloadChoices) {
+        auto *refExpr = pair.first;
+        auto *decl = pair.second;
+
+        refExpr->setVariable(decl);
+    }
+}
+
+void ConstraintSystem::mapExprTypes(Solution *solution)
+{
+    for (auto &pair : solution->exprTypes) {
+        auto *expr = pair.first;
+        auto *type = pair.second;
+
+        expr->setType(type);
+    }
+}
+
+void ConstraintSystem::mapImplicitConversions(Solution *solution)
+{
+    for (auto &pair : solution->implicitConversions) {
+        auto *expr = pair.first;
+        auto *targetType = pair.second;
+        auto *castExpr = _context->getASTMemoryArena().create<ast::CastExpr>(
+            expr->getLocation(), expr, targetType
+        );
+
+        castExpr->setParent(expr->getParent());
+        expr->setParent(castExpr);
```

Comment:
expr parent is already set
you have to replace the child however in expr->getParent(), which might not be easy (probably need a visitor to check children)